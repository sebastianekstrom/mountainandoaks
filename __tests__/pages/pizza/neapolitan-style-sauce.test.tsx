import React from "react";
import { render, screen } from "@testing-library/react";
import NeopolitanSauce from "pages/pizza/neapolitan-style-sauce";

describe("NeopolitanSauce page", () => {
  it("renders title correctly", () => {
    render(<NeopolitanSauce />);
    const title = screen.queryAllByText(/Neapolitan style pizza sauce/i);
    expect(title).toBeDefined();
  });

  it("renders the ingress text", () => {
    render(<NeopolitanSauce />);
    const ingress = screen.queryAllByText(
      /There's a lot of variants for the tomato sauce/i
    );
    expect(ingress).toBeDefined();
  });

  it("renders the ingredients", () => {
    render(<NeopolitanSauce />);
    const flourLabel = screen.queryAllByText(/Canned San Marzano tomatoes/i);
    const waterLabel = screen.queryAllByText(/Leaves of fresh basil/i);
    expect(flourLabel).toBeDefined();
    expect(waterLabel).toBeDefined();
  });

  it("renders the method steps", () => {
    render(<NeopolitanSauce />);
    const step = screen.queryAllByText(/When the pan has heated up, add the/i);
    expect(step).toBeDefined();
  });
});
