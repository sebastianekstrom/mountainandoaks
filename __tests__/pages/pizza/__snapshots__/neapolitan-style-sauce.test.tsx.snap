// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NeopolitanSauce page renders the correct ingredients 1`] = `
<section
  class="mb-8 lg:mb-0"
  data-testid="INGREDIENT_LIST"
>
  <div>
     
  </div>
  <div
    class="flex flex-col"
  >
    <ul>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Canned San Marzano tomatoes
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          400
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Olive oil
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          15
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Sugar
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          4.5
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Salt
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          4.5
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Leaves of fresh basil
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          10
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Clove of garlic
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          1
        </p>
      </li>
    </ul>
  </div>
</section>
`;
